src/repository/opamRepositoryPath.ml

```diff
@@ -15,6 +15,19 @@ let create root name = root / "repo" / OpamRepositoryName.to_string name
 
 let download_cache root = root / "download-cache"
 
+let pin_cache_dir =
+  let dir =
+    lazy (OpamFilename.Dir.of_string (OpamStd.Sys.tmp ()) /
+          (Printf.sprintf "opam-pin-cache-%d" @@ OpamStubs.getpid ()))
```

Comment:
`OpamSystem.temp_basename "opam-pin-cache"` ? Or even `OpamSystem.mk_temp_dir ()` ?
Either of which would need to be exported.
The added random bits would mitigate /tmp race attacks.